docs/docsite/rst/user_guide/playbooks_intro.rst

```diff
@@ -405,6 +405,23 @@ Here's an example handlers section::
             name: apache
             state: restarted
 
+Avoid using variables in handler names. Since handler names are templated early on, all variables may not be available, resulting in play failure. It is possible to use variables in handler task parameters. For example, if the name of a service varies slightly by distribution, the service name can be set using ``include_vars`` (or any place where variables can be set).
```

Comment:
I wonder if we should clarify that in case users choose to use variables in handler names then changing that variable does not result in some kind of magic dynamic creation of new handlers, which people thought in that issue it seemed.

Reply:
diff name == diff handler, so that would be a specific feature of doing this